source/as-promise.ts

```diff
@@ -27,7 +27,7 @@ const parseBody = (body: Response['body'], responseType: NormalizedOptions['resp
 	throw new Error(`Failed to parse body of type '${typeof body}' as '${responseType}'`);
 };
 
-export default function asPromise<T extends any>(options: NormalizedOptions): CancelableRequest<T> {
+export default function asPromise<T extends unknown = unknown>(options: NormalizedOptions): CancelableRequest<T> {
```

Comment:
I don't think you need to constrain these to `unknown`. The default should be enough: https://devblogs.microsoft.com/typescript/announcing-typescript-3-5/#breaking-changes I think it could be just `<T>(..`. Same in the other places we have this.